### PR TITLE
Avoid redundant reads for FK checks in NewOrder

### PR DESCRIPTION
**Pull request by d-uspenskiy**

### CODE REVIEW

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 1]
```diff
@@ -182,7 +182,7 @@ public NewOrder() {
       } else {
         sb.append(",?");
       }
-      stmtGetStockSQLArr[i - 1] = new InstrumentedSQLStmt(stmtGetStockSQL.getHistogram(), sb.toString() + ")");
+      stmtGetStockSQLArr[i - 1] = new InstrumentedSQLStmt(stmtGetStockSQL.getHistogram(), sb.toString() + ") FOR KEY SHARE");
```
Comment (robertsami) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java:185]: we should add this clause in the prepared statement, not here, if possible

do you know why we were adding the right hand parens?

Comment (d-uspenskiy) [reply src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java:185]: @robertsami could you please clarify the question?

the `stmtGetStockSQLArr` array stores queries which are used to build prepared statements later at line https://github.com/yugabyte/tpcc/blob/master/src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java#L410

**Comment by hbhanawat** [comment]
@d-uspenskiy if the server side doesn't have your changes, will this change degrade the performance?

**Comment by d-uspenskiy** [comment]
@hbhanawat In case YB doesn't have changes with the optimization of FK check this changes in TPC-C client will decrease the performance a little bit.
The reason is:
Before this change TPC-C does 2 read operations `SELECT` + `FK check`. In this case YB's postgres process produces 2 read RPCs to DocDB. The first read RPC (for `SELECT`) has no `ROW MARK`, as a result it doesn't produces any intents on DocDB side. Only second read RPC (for `FK check`) will have `ROW MARK` and will produces intents on DocDB side.

After the current change TPC-C client still does 2 operations `SELECT with ROW MARK` + `FK check`. But YB's postgres (if it has all the optimizations) will generate single read RPC only for (`SELECT with ROW MARK`). `FK check` read operations will be omitted. As a result DocDB will receive single read RPC with `ROW MARK` which is better that 2 read RPCs.

But in case YB's postgres process has no optimization for `FK check` it will send 2 read RPCs to DocDB and both these RPCs will has `ROW MARK` and will create intents. From the DocDB side 2 RPCs both with `ROW MARK` requires more computations than 2 RPCs where only one has  `ROW MARK`.

**Comment by hbhanawat** [comment]
@d-uspenskiy Thanks for the detailed description. Are you backporting your change to previous releases? If not, we should wait for sometime before merging this to master.

**Review by robertsami — APPROVED** [review]

**Pull request merged**